modules/stochastic_tools/include/vectorpostprocessors/StochasticResults.h

```diff
@@ -55,18 +56,14 @@ class StochasticResults : public GeneralVectorPostprocessor, SamplerInterface
   void setCurrentLocalVectorPostprocessorValue(const std::string & vector_name,
                                                const VectorPostprocessorValue && current);
 
-  /**
-   * DEPRECATED
-   * Initialize storage based on the Sampler returned by the SamplerTransientMultiApp or
-   * SamplerFullSolveMultiApp.
-   * @param sampler The Sampler associated with the MultiApp that this VPP is working with.
-   *
-   * This an internal method that is called by the SamplerPostprocessorTransfer, it should not
-   * be called otherwise.
-   */
-  void init(Sampler & _sampler);
-
 protected:
   /// Storage for declared vectors
   std::vector<StochasticResultsData> _sample_vectors;
+
+  /**
+   * Create a VPP vector for results data for a given Sampler, see StochasticResultsAction for
+   * more details to why this is necessary.
+   */
+  void initVector(const std::string & vector_name);
+  friend void StochasticResultsAction::act();
```

Comment:
would be nice to have a comment for why this is necessary

Reply:
See line 64

Reply:
Lol. My bad